lib/action_controller/serialization.rb

```diff
@@ -60,7 +60,15 @@ def serialization_scope(scope)
     private
 
     def namespace_for_serializer
-      @namespace_for_serializer ||= self.class.parent unless self.class.parent == Object
+      @namespace_for_serializer ||= namespace_for_class(self.class) unless namespace_for_class(self.class) == Object
+    end
+
+    def namespace_for_class(klass)
+      if Module.method_defined?(:module_parent)
+        klass.module_parent
+      else
+        klass.parent
+      end
```

Comment:
hmm, my own preference for dynamic Ruby code is if something can be dynamically evaluated at load, there's no reason to evaluate again.

i.e. 

```diff
+    if Module.method_defined?(:module_parent)
+    def namespace_for_class(klass)
+        klass.module_parent
+    end
+    else
+    def namespace_for_class(klass)
+        klass.parent
+    end
+    end
-    def namespace_for_class(klass)
-      if Module.method_defined?(:module_parent)
-        klass.module_parent
-      else
-        klass.parent
-      end
```

otoh, it looks like this was just copied from 0-10-stable

Reply:
Yeah, I figured using the same code as 0-10-stable would help with maintenance 🤷‍♂

Reply:
0-9 and 0-10 are very different in features and code. I think it is better to go with the conditional at load time instead of runtime.

Reply:
I will try to go with the **load time** logic on **0-10** some day